xarray/core/variable.py

```diff
@@ -58,14 +59,21 @@
     maybe_coerce_to_str,
 )
 
-NON_NUMPY_SUPPORTED_ARRAY_TYPES = (
-    (
-        indexing.ExplicitlyIndexed,
-        pd.Index,
+
+def _get_NON_NUMPY_SUPPORTED_ARRAY_TYPES():
+    """Required instead of a global to avoid circular import errors with cubed"""
+
+    return (
+        (
+            indexing.ExplicitlyIndexed,
+            pd.Index,
+        )
+        + dask_array_type
+        + cupy_array_type
+        + DuckArrayModule("cubed").type
     )
```

Comment:
90% sure we can delete explicit support for dask and cupy (and so not need to add cubed), they should be following standard array protocols in the Variable constructor now. Can you send in a separate PR to remove this?

Reply:
Opened #7023 for this